drivers/adc/adc_ite_it8xxx2.c

```diff
@@ -310,9 +334,23 @@ static struct adc_it8xxx2_data adc_it8xxx2_data_0 = {
 		ADC_CONTEXT_INIT_LOCK(adc_it8xxx2_data_0, ctx),
 		ADC_CONTEXT_INIT_SYNC(adc_it8xxx2_data_0, ctx),
 };
+static const struct adc_it8xxx2_cfg adc_it8xxx2_cfg_0 = {
```

Comment:
I think this would be clearer if struct adc_it8xxx2_cfg defined a single ADC channel only, and then adc_it8xxx_cfg_0 is defined as an array.

struct adc_it8xxx2_cfg {
	/* Pinmux control group */
	const struct device *pinctrls;
	/* GPIO pin */
	uint8_t pin
	/* Alternate function */
	uint8_t alt_fun
};
static const struct adc_it8xxx2_cfg adc_it8xxx2_cfg_0[8] = {
    [0] = { .pin = DEV_PIN(0), .alt_func = DEV_ALT_FUNC(0), .pinctrls  = DEV_PINMUX(0), },
    [1] = ...
};

You can probably also use UTIL_LISTIFY to create all the ADC entries.

Reply:
All the ADC entries are be modified to use UTIL_LISTIFY,